### PR TITLE
[DOCS] Bring code examples inline with style guide in assembly doc

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Description

As part of #4580 and #4581 this PR fixes code examples in assembly doc that violate the style guide.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

File: docs/assembly.rst [hunk 1]
```diff
@@ -104,47 +104,48 @@ efficient code, for example:
...
-
-               // Iterate until the bound is not met.
-               for
-                   { let end := add(data, mul(len, 0x20)) }
-                   lt(data, end)
-                   { data := add(data, 0x20) }
-               {
-                   o_sum := add(o_sum, mload(data))
-               }
+            // Load the length (first 32 bytes)
```
Comment (leonardoalt) [inline docs/assembly.rst]: Where does the style guide mention that inline assembly blocks shouldn't use indentation? I just searched by couldn't find it

Comment (ChrisChinchilla) [reply docs/assembly.rst]: Ahh, I see what you mean. That might be an issue with Solhint, I'll investigate

Comment (ChrisChinchilla) [reply docs/assembly.rst]: Fixed it here, but looks like solhint misses that

File: docs/assembly.rst [hunk 2]
```diff
@@ -104,47 +104,48 @@ efficient code, for example:
 
     pragma solidity >=0.4.16 <0.7.0;
 
+
     library VectorSum {
         // This function is less efficient because the optimizer currently fails to
         // remove the bounds checks in array access.
-        function sumSolidity(uint[] memory _data) public pure returns (uint o_sum) {
```
Comment (chriseth) [inline docs/assembly.rst:110]: Why not just call it `sum`?

Comment (ChrisChinchilla) [reply docs/assembly.rst:110]: Seems fine to me! I have no idea who named it `o_sum` or `oSum` and if there was a reason for it, but it seems reasonable var name to me.

**Review by chriseth — APPROVED** [review]

**Pull request merged**